omnidocs/tasks/layout_analysis/extractors/rtdetr.py

```
@@ -202,78 +235,62 @@ def detect(
-                img = img.cpu()
-                orig_size = orig_size.cpu()
+                inputs = {k: v.cuda() if isinstance(v, torch.Tensor) else v for k, v in inputs.items()}
 
-            # Run inference with error handling
+            # Run inference
             try:
                 with torch.no_grad():
-                    labels, boxes, scores = self.model(img, orig_size)
+                    outputs = self.model(**inputs)
             except Exception as e:
                 raise RuntimeError(f"Error during model inference: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)